org.eclipse.xtext/src/org/eclipse/xtext/build/Indexer.java

```diff
@@ -213,21 +221,32 @@ public URI getURI() {
 	protected List<IResourceDescription.Delta> getDeltasForDeletedResources(BuildRequest request,
 			ResourceDescriptionsData oldIndex, BuildContext context) {
 		List<IResourceDescription.Delta> deltas = new ArrayList<>();
-		for (URI deleted : request.getDeletedFiles()) {
-			IResourceServiceProvider resourceServiceProvider = context.getResourceServiceProvider(deleted);
-			if (resourceServiceProvider != null) {
-				operationCanceledManager.checkCanceled(context.getCancelIndicator());
-				IResourceDescription oldDescription = oldIndex != null ? oldIndex.getResourceDescription(deleted)
-						: null;
-				if (oldDescription != null) {
-					DefaultResourceDescriptionDelta delta = new DefaultResourceDescriptionDelta(oldDescription, null);
-					deltas.add(delta);
+		if (oldIndex != null) {
```

Comment:
I have pushed the null check on the oldIndex outside of the loop, but I am not sure it can ever happen that it will be null, if it would be null, the loop I have modified (line 204 on this PR) would throw an NPE. So I think we could remove it as well